### PR TITLE
Feature message_service

### PR DESCRIPTION
**Pull request by xandao6**
## Proposed changes

It's a substitute for the API, allow you to communicate your programs to wplay. You modify our file messages.json with the number and message and the service will read the JSON and send the message.
I also fixed the alerts that appeared and prevented the broadcast from working.

## Types of changes

- [X] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

## Screenshots

image

### CODE REVIEW

File: wplay/utils/target_search.py [hunk 1]
```diff
@@ -143,10 +144,22 @@ async def search_and_select_target_without_new_chat_button(page: Page, target: s
 
 
 # region SEARCH AND SELECT TARGET
+async def accept_dialog(dialog):
+    try:
+        await dialog.accept()
+    except:
```
Comment (rpotter12) [inline wplay/utils/target_search.py:150]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Complexity increasing per file
==============================
- wplay/utils/MessageStack.py  4
- wplay/message_service.py  7
- wplay/utils/verify_internet.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804&bid=17616001)

File: wplay/utils/verify_internet.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import http.client as httplib
+
+def internet_avalaible():
+    conn = httplib.HTTPConnection("www.google.com", timeout=5)
+    try:
+        conn.request("HEAD", "/")
+        conn.close()
+        return True
+    except:
```
Comment (rpotter12) [inline wplay/utils/verify_internet.py:9]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        json_data['messages'].append(message)
+        
+        self.__write_json(json_data, file_path)
+        self.logger.info(f'Message appended to {file_path.name}')
+
+    def get_message(
+            self,
+            from_file_path: Path = helpers.messages_json_path) -> Iterator[dict]:
+        """
+        Yield a message from a file.
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:59]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# region IMPORTS
+from pathlib import Path
+import threading
+import time
+import json
+
+from wplay.utils import browser_config
+from wplay.utils import target_search
+from wplay.utils import target_select
```
Comment (rpotter12) [inline wplay/message_service.py:9]: Codacy Issue found: [Unused target_select imported from wplay.utils](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        self.__ensure_valid_json(file_path)
+
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+        json_data['messages'].append(message)
+        
+        self.__write_json(json_data, file_path)
+        self.logger.info(f'Message appended to {file_path.name}')
+
+    def get_message(
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:55]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# region IMPORTS
+from pathlib import Path
+import threading
```
Comment (rpotter12) [inline wplay/message_service.py:3]: Codacy Issue found: [Unused import threading](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        self.__ensure_valid_json(helpers.open_messages_json_path)
+
+    def __create_json_file(self, file_path: Path):
+        if not file_path.is_file():
+            open(file_path, 'w').close()
+            self.logger.info(f'{file_path.name} created.')
+
+    def __write_json(self, data: dict, file_path: Path):
+        with open(file_path, "w") as json_file:
+            json.dump(data, json_file, indent=4)
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:25]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        self.__ensure_valid_json(helpers.messages_json_path)
+        self.__create_json_file(helpers.open_messages_json_path)
+        self.__ensure_valid_json(helpers.open_messages_json_path)
+
+    def __create_json_file(self, file_path: Path):
+        if not file_path.is_file():
+            open(file_path, 'w').close()
+            self.logger.info(f'{file_path.name} created.')
+
+    def __write_json(self, data: dict, file_path: Path):
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:23]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        Arguments:
+            message -- dict with a message
+            file_path {Path} -- open_messages_json_path or messages_json_path from helpers
+        """
+        self.__ensure_valid_json(file_path)
+
+        with open(file_path) as json_file:
+            json_data = json.load(json_file)
+        json_data['messages'].append(message)
+
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+        with open(file_path, "w") as json_file:
+            json.dump(data, json_file, indent=4) 
+
+    def __ensure_valid_json(self, file_path: Path):
+        valid_data = {"messages":list()}
+        try:
+            with open(file_path) as json_file:
+                data = json.load(json_file)
+                if not 'messages' in data:
+                    self.__write_json(valid_data, file_path)
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:33]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+                    current_msg['uuid'])
+
+                try:
+                    try:
+                        await target_search.search_and_select_target(page, current_msg['number'])
+                    except Exception as e:
+                        __logger.debug("Error searching the target, ensure that is a full number with country code. Message Deleted")
+                        message_stack.remove_message(
+                            current_msg['uuid'], helpers.open_messages_json_path)
+
```
Comment (rpotter12) [inline wplay/message_service.py:74]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+            for i, message in enumerate(data['messages']):
+                if uuid in message['uuid']:
+                    del data['messages'][i]
+                    self.__write_json(data, file_path)
+                    self.logger.info(f"Message Deleted Successfully from {file_path.name}")
+
+    def remove_all_messages(self, file_path: Path):
+        with open(file_path) as json_file:
+            data = json.load(json_file)
+            for i, message in enumerate(data['messages']):
```
Comment (rpotter12) [inline wplay/utils/MessageStack.py:111]: Codacy Issue found: [Unused variable 'message'](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 12
           

Complexity increasing per file
==============================
- wplay/utils/verify_internet.py  2
- wplay/message_service.py  7
- wplay/utils/MessageStack.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804&bid=17616001)

File: wplay/utils/MessageStack.py (hunk 3 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:59]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 12 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:111]: Codacy Issue found: [Unused variable 'message'](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/target_search.py (hunk 1 above)
Comment (rpotter12) [inline wplay/utils/target_search.py:150]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py (hunk 4 above)
Comment (rpotter12) [inline wplay/message_service.py:9]: Codacy Issue found: [Unused target_select imported from wplay.utils](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py (hunk 6 above)
Comment (rpotter12) [inline wplay/message_service.py:3]: Codacy Issue found: [Unused import threading](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py (hunk 11 above)
Comment (rpotter12) [inline wplay/message_service.py:74]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 7 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:25]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 8 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:23]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 10 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:33]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 9 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/verify_internet.py (hunk 2 above)
Comment (rpotter12) [inline wplay/utils/verify_internet.py:9]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 5 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:55]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- wplay/utils/MessageStack.py  4
- wplay/utils/verify_internet.py  2
- wplay/message_service.py  7
         

Complexity decreasing per file
==============================
+ wplay/broadcast_message.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804&bid=17616001)

File: wplay/utils/verify_internet.py (hunk 2 above)
Comment (rpotter12) [inline wplay/utils/verify_internet.py:9]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 12 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:111]: Codacy Issue found: [Unused variable 'message'](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 5 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:55]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py [hunk 13]
```diff
@@ -0,0 +1,92 @@
+# region IMPORTS
+from pathlib import Path
+import threading
```
Comment (rpotter12) [inline wplay/message_service.py:3]: Codacy Issue found: [Unused import threading](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+import time
+import json
+
+from wplay.utils import browser_config
+from wplay.utils.target_search import search_target_by_number
+from wplay.utils import target_select
```
Comment (rpotter12) [inline wplay/message_service.py:9]: Codacy Issue found: [Unused target_select imported from wplay.utils](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 10 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:33]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 8 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:23]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/target_search.py [hunk 15]
```diff
@@ -143,11 +150,29 @@ async def search_and_select_target_without_new_chat_button(page: Page, target: s
 
 
 # region SEARCH AND SELECT TARGET
+async def __accept_dialog(dialog):
+    try:
+        await dialog.accept()
+    except:
```
Comment (rpotter12) [inline wplay/utils/target_search.py:156]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 3 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:59]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 7 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:25]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 9 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

**Comment by xandao6** [comment]
I did a rebase.

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- wplay/utils/MessageStack.py  4
- wplay/message_service.py  7
- wplay/__main__.py  1
- wplay/utils/verify_internet.py  2
         

Complexity decreasing per file
==============================
+ wplay/broadcast_message.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804&bid=17616001)

File: wplay/utils/MessageStack.py (hunk 12 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:111]: Codacy Issue found: [Unused variable 'message'](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 7 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:25]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 5 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:55]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 9 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 10 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:33]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/verify_internet.py (hunk 2 above)
Comment (rpotter12) [inline wplay/utils/verify_internet.py:9]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 8 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:23]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py (hunk 14 above)
Comment (rpotter12) [inline wplay/message_service.py:9]: Codacy Issue found: [Unused target_select imported from wplay.utils](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/MessageStack.py (hunk 3 above)
Comment (rpotter12) [inline wplay/utils/MessageStack.py:59]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/utils/target_search.py (hunk 15 above)
Comment (rpotter12) [inline wplay/utils/target_search.py:156]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

File: wplay/message_service.py (hunk 13 above)
Comment (rpotter12) [inline wplay/message_service.py:3]: Codacy Issue found: [Unused import threading](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5373804)

**Pull request merged**